src/App.tsx

```diff
@@ -1,24 +1,245 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { UserWarning } from './components/UserWarning/UserWarning';
 
-const USER_ID = 0;
+import {
+  getTodos,
+  postTodo,
+  deleteTodo,
+  patchTodo,
+} from './api/todos';
+
+import { Todo } from './types/Todo';
+import { FilterType } from './utils/filterTypes';
+
+import { Loader } from './components/Loader/Loader';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+
+const USER_ID = 6979;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState('');
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [isDisabledInput, setIsDisabledInput] = useState(false);
+  const [loadingIds, setLoadingIds] = useState<number[]>([]);
+
+  const clearError = () => {
+    setTimeout(() => {
+      setError('');
+    }, 3000);
+  };
+
+  const loadTodos = async () => {
+    setIsLoading(true);
+
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch {
+      setError('Unable to load a todo');
+
+      clearError();
+
+      setTodos([]);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const addTodo = async (title: string) => {
+    setIsDisabledInput(true);
+
+    const newTodo = {
+      title,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    setTempTodo({ ...newTodo, id: 0 });
+
+    try {
+      const data = await postTodo(newTodo);
+
+      setTodos(state => [...state, data]);
+    } catch {
+      setError('Unable to add a todo');
+      clearError();
+    } finally {
+      setIsDisabledInput(false);
+      setTempTodo(null);
+    }
+  };
+
+  const removeTodo = async (id: number) => {
+    setLoadingIds(state => [...state, id]);
+
+    try {
+      await deleteTodo(id);
+
+      setTodos(() => todos.filter(todo => todo.id !== id));
+    } catch {
+      setError('Unable to delete a todo');
+      clearError();
+    } finally {
+      setLoadingIds(state => state.filter(el => el !== id));
+    }
+  };
+
+  const removeCompleted = () => {
+    const completed = todos.filter(todo => todo.completed);
+
+    completed.forEach(todo => {
+      removeTodo(todo.id)
+        .then(() => setTodos(todos.filter(item => !item.completed)))
+        .catch(() => setError('Unable to delete todos'));
```

Comment:
You are calling delete requests for many Todos, and some may fail while others will work. It's not correct to write 'Unable to delete todos' because this failure is related to one specific todo.